### ISSUE TITLE
Restored bosses need to be updated to Glad Giraffe monster framework

### ISSUE THREAD

**Issue by dsp2003** [dsp2003/clef#4]
CF has suddenly rewrote all monster behaviour libraries into single monster.lua framework. This lead to complete breakage of all old boss monster code.

CLeF is not going to be Stable Glad Giraffe-compatible without working bosses. :(

**Comment by dsp2003** [comment]
Not just bosses, it appears all monsters, even critters and pets, will need to be ported. Urgh... /)_<

**Comment by dsp2003** [comment]
@LoPhatKao , I summon thou. (\(@Д@)/)

**Comment by LoPhatKao** [comment]
so i really hate new monster code... ;(
will take a look at this 
(can tell right away sound code is fubar, looked at it a while ago for screams mod)

**Comment by dsp2003** [comment]
@LoPhatKao I've tried to go the different approach - copy old scripts and skills from PG (because they removed every damn thing!). Of course I've met the name collision errors, so tried renaming some of them - the game stopped crashing, yet the monsters still were dying. A frustrating experience indeed. :(

**Comment by Joshwoo69** [comment]
everything now ported over to glad giraffe?

**Comment by dsp2003** [comment]
I'm afraid, no. LoPhatKao still hasn't committed any code regarding the monsters. The problems are laying within too much (and ugly) under-the-hood changes CF did to monster framework. Unless someone is willing to rewrite the code, there won't be working old bosses (at least, not with the previously possible functionality).

**Comment by Joshwoo69** [comment]
So.. everything in the mod works... except the bosses... okay.

**Comment by LoPhatKao** [comment]
sorry :| i got really distracted by vehicles ;)
on the plus side, i made XS Mechs work in Glad & Cheerful Giraffes :)
starting work on this now, might see a way to fix fast (i hope heh)

**Comment by dsp2003** [comment]
OH MY GOSH, FINALLY! :D

**Comment by LoPhatKao** [comment]
have dragon working fine, fatal circuit is quite.. fatal :|

**Comment by dsp2003** [comment]
@LoPhatKao , it appears they have removed all skills from the randomly generated monsters that Fatal Circuit's code was based on. I've tried to put them back and link the monster to old "framework" instead (and also renamed duplicate skills, both in filenames and internally), but it gave out the Lua exceptions. :(

**Comment by LoPhatKao** [comment]
yeah think im gonna need to rewrite groundmonster based bosses, keeps crashing no matter what i try
even took all the luas from pleased and had it only try to use that.. worked for dragon, wont for ground

**Comment by LoPhatKao** [comment]
1 down, several to go \o/
